junit5/src/main/java/cz/xtf/junit5/model/DockerImageMetadata.java

```diff
@@ -161,4 +165,42 @@ public String command() {
 
 		return result;
 	}
+
+	private static final class ImageStreamFailFastCheck implements FailFastCheck{
+
+		private final OpenShift openShift;
+		private final String imageName;
+		private final Image image;
+		private String reason = "";
+
+		ImageStreamFailFastCheck(OpenShift openShift, String imageName, Image image) {
+			this.openShift = openShift;
+			this.imageName = imageName;
+			this.image = image;
+		}
+
+		@Override
+		public boolean hasFailed() {
+			ImageStream imageStream = openShift.getImageStream(imageName);
+			if (imageStream == null) {
+				return false;
+			}
+			for (NamedTagEventList tag : imageStream.getStatus().getTags()) {
+				if (image.getTag().startsWith(tag.getTag())) {
+					for (TagEventCondition condition : tag.getConditions()) {
+						if (condition.getType().equals("ImportSuccess") && condition.getStatus().equals("False")) {
```

Comment:
In case that image stream is "creation in progress" then `ImportSuccess` is undefined? I'm just worrying how it works and would like to avoid issues when imagestream creation is in progress and this value being `false`. It would fail instead of wait.

Reply:
No. Such list (conditions) is empty when import is successful. Even during the operation.

Reply:
ok, thanks